### ISSUE TITLE
[BUG] master don't finde gitpyhton, but its installed

### ISSUE THREAD

**Issue by colttt** [saltstack/salt#64504]
**Description**
Since we updated our salt master to 3006.1, it doesn't find the gitpython

**Setup**
(Please provide relevant configs and/or SLS files (be sure to remove sensitive info. There is no general set-up of Salt.)

Please be as specific as possible and give set-up details.

- [ ] on-prem machine
- [x] VM VMware
- [ ] VM running on a cloud service, please be explicit and add details
- [ ] container (Kubernetes, Docker, containerd, etc. please specify)
- [ ] or a combination, please be explicit
- [ ] jails if it is FreeBSD
- [ ] classic packaging
- [x] onedir packaging
- [ ] used bootstrap to install

**Steps to Reproduce the behavior**
we just run `apt update && apt upgrade`
```
 salt-master -l debug
[DEBUG   ] Reading configuration from /etc/salt/master
[DEBUG   ] Including configuration from '/etc/salt/master.d/gitfs.conf'
[DEBUG   ] Reading configuration from /etc/salt/master.d/gitfs.conf
[DEBUG   ] Including configuration from '/etc/salt/master.d/output.conf'
[DEBUG   ] Reading configuration from /etc/salt/master.d/output.conf
[DEBUG   ] Changed git to gitfs in master opts' fileserver_backend list
[DEBUG   ] Using cached minion ID from /etc/salt/minion_id: salt.aei.mpg.de
[WARNING ] Insecure logging configuration detected! Sensitive data may be logged.
[DEBUG   ] Configuration file path: /etc/salt/master
[INFO    ] Setting up the Salt Master
[DEBUG   ] Marking 'base64_encode' as a jinja filter
[DEBUG   ] Marking 'base64_decode' as a jinja filter
[DEBUG   ] Marking 'md5' as a jinja filter
[DEBUG   ] Marking 'sha1' as a jinja filter
[DEBUG   ] Marking 'sha256' as a jinja filter
[DEBUG   ] Marking 'sha512' as a jinja filter
[DEBUG   ] Marking 'hmac' as a jinja filter
[DEBUG   ] Marking 'hmac_compute' as a jinja filter
[DEBUG   ] Marking 'random_hash' as a jinja filter
[DEBUG   ] Marking 'rand_str' as a jinja filter
[DEBUG   ] Marking 'file_hashsum' as a jinja filter
[DEBUG   ] Marking 'http_query' as a jinja filter
[DEBUG   ] Marking 'ifelse' as a jinja global
[DEBUG   ] Marking 'strftime' as a jinja filter
[DEBUG   ] Marking 'date_format' as a jinja filter
[DEBUG   ] Marking 'raise' as a jinja global
[DEBUG   ] Marking 'match' as a jinja test
[DEBUG   ] Marking 'equalto' as a jinja test
[DEBUG   ] Marking 'skip' as a jinja filter
[DEBUG   ] Marking 'sequence' as a jinja filter
[DEBUG   ] Marking 'to_bool' as a jinja filter
[DEBUG   ] Marking 'indent' as a jinja filter
[DEBUG   ] Marking 'tojson' as a jinja filter
[DEBUG   ] Marking 'quote' as a jinja filter
[DEBUG   ] Marking 'regex_escape' as a jinja filter
[DEBUG   ] Marking 'regex_search' as a jinja filter
[DEBUG   ] Marking 'regex_match' as a jinja filter
[DEBUG   ] Marking 'regex_replace' as a jinja filter
[DEBUG   ] Marking 'uuid' as a jinja filter
[DEBUG   ] Marking 'unique' as a jinja filter
[DEBUG   ] Marking 'min' as a jinja filter
[DEBUG   ] Marking 'max' as a jinja filter
[DEBUG   ] Marking 'avg' as a jinja filter
[DEBUG   ] Marking 'union' as a jinja filter
[DEBUG   ] Marking 'intersect' as a jinja filter
[DEBUG   ] Marking 'difference' as a jinja filter
[DEBUG   ] Marking 'symmetric_difference' as a jinja filter
[DEBUG   ] Marking 'method_call' as a jinja filter
[DEBUG   ] Marking 'yaml_dquote' as a jinja filter
[DEBUG   ] Marking 'yaml_squote' as a jinja filter
[DEBUG   ] Marking 'yaml_encode' as a jinja filter
[DEBUG   ] Popen(['git', 'version'], cwd=/root, universal_newlines=False, shell=None, istream=None)
[DEBUG   ] salt.crypt.get_rsa_key: Loading private key
[DEBUG   ] salt.crypt._get_key_with_evict: Loading private key
[DEBUG   ] Loaded master key: /etc/salt/pki/master/master.pem
[INFO    ] Preparing the salt key for local communication
[DEBUG   ] Removing stale keyfile: /var/cache/salt/master/.salt_key
[INFO    ] Preparing the root key for local communication
[DEBUG   ] Removing stale keyfile: /var/cache/salt/master/.root_key
[PROFILE ] Beginning pwd.getpwall() call in masterapi access_keys function
[PROFILE ] End pwd.getpwall() call in masterapi access_keys function
[DEBUG   ] Created pidfile: /var/run/salt-master.pid
[DEBUG   ] Chowned pidfile: /var/run/salt-master.pid to user: salt
[INFO    ] Starting up the Salt Master
[ERROR   ] gitfs is configured but could not be loaded, is GitPython installed?
[CRITICAL] No suitable gitfs provider module is installed.
[DEBUG   ] The functions from module 'roots' are being loaded by dir() on the loaded module
[DEBUG   ] The functions from module 's3fs' are being loaded by dir() on the loaded module
[DEBUG   ] Could not LazyLoad gitfs.envs: 'gitfs' __virtual__ returned False
[ERROR   ] git_pillar is configured but could not be loaded, is GitPython installed?
[CRITICAL] No suitable git_pillar provider module is installed.
[CRITICAL] Failed to load git_pillar
[CRITICAL] Master failed pre flight checks, exiting
```

also salt master said its not installed:

```
salt --versions-report
Salt Version:
          Salt: 3006.1
 
Python Version:
        Python: 3.10.11 (main, May  5 2023, 02:31:54) [GCC 11.2.0]
 
Dependency Versions:
          cffi: 1.14.6
      cherrypy: unknown
      dateutil: 2.8.1
     docker-py: Not Installed
         gitdb: 4.0.10
     gitpython: Not Installed
        Jinja2: 3.1.2
       libgit2: Not Installed
  looseversion: 1.0.2
      M2Crypto: Not Installed
          Mako: Not Installed
       msgpack: 1.0.2
  msgpack-pure: Not Installed
  mysql-python: Not Installed
     packaging: 22.0
     pycparser: 2.21
      pycrypto: Not Installed
  pycryptodome: 3.9.8
        pygit2: Not Installed
  python-gnupg: 0.4.8
        PyYAML: 5.4.1
         PyZMQ: 23.2.0
        relenv: 0.12.3
         smmap: 5.0.0
       timelib: 0.2.4
       Tornado: 4.5.3
           ZMQ: 4.3.4
 
System Versions:
          dist: debian 11 bullseye
        locale: utf-8
       machine: x86_64
       release: 5.10.0-23-amd64
        system: Linux
       version: Debian GNU/Linux 11 bullseye
 
```

BUT pip list said it is..
```
salt-pip list
Package            Version
------------------ ---------
certifi            2022.12.7
cffi               1.14.6
chardet            3.0.4
cheroot            8.5.2
CherryPy           18.6.1
contextvars        2.4
cryptography       39.0.2
distro             1.5.0
gitdb              4.0.10
GitPython          3.1.31
idna               2.8
immutables         0.15
importlib-metadata 6.0.0
jaraco.classes     3.2.1
jaraco.collections 3.4.0
jaraco.functools   2.0
jaraco.text        3.5.1
Jinja2             3.1.2
jmespath           1.0.1
looseversion       1.0.2
MarkupSafe         2.1.2
more-itertools     5.0.0
msgpack            1.0.2
packaging          22.0
pip                23.1.2
portend            2.4
psutil             5.8.0
pycparser          2.21
pycryptodomex      3.9.8
pyOpenSSL          23.0.0
python-dateutil    2.8.1
python-gnupg       0.4.8
pytz               2022.1
PyYAML             5.4.1
pyzmq              23.2.0
relenv             0.12.3
requests           2.25.1
rpm-vercmp         0.1.2
salt               3006.1
setproctitle       1.3.2
setuptools         65.7.0
six                1.16.0
smmap              5.0.0
tempora            4.1.1
timelib            0.2.5
urllib3            1.26.6
wheel              0.40.0
zc.lockfile        1.4
zipp               3.6.0
```
and
```
salt-pip install gitpython
Requirement already satisfied: gitpython in /opt/saltstack/salt/extras-3.10 (3.1.31)
Requirement already satisfied: gitdb<5,>=4.0.1 in /opt/saltstack/salt/extras-3.10 (from gitpython) (4.0.10)
Requirement already satisfied: smmap<6,>=3.0.1 in /opt/saltstack/salt/extras-3.10 (from gitdb<5,>=4.0.1->gitpython) (5.0.0)
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
```

**Expected behavior**
a working salt-master which finds the gitpython

**Comment by OrangeDog** [comment]
https://docs.saltproject.io/salt/install-guide/en/latest/topics/upgrade-to-onedir.html

**Comment by ssoto2** [comment]
Why was this closed? What is the resolution? 

@colttt specifically says he performed 

'''
salt-pip install gitpython
'''

and yet it is not working. The documentation doesn't even say hey install gitpython through salt-pip, but they still did it and having errors. what is the fix?

**Comment by st-claude** [comment]
i have the exact same problem with a clean fresh install following official doc on debian 12

```
root@salt:/usr/bin# salt-pip list|grep Git
GitPython          3.1.41

root@salt:/usr/bin# salt --versions-report
Salt Version:
          Salt: 3006.5

Python Version:
        Python: 3.10.13 (main, Nov 15 2023, 04:34:27) [GCC 11.2.0]

Dependency Versions:
          cffi: 1.14.6
      cherrypy: unknown
      dateutil: 2.8.1
     docker-py: Not Installed
         gitdb: 4.0.11
     gitpython: Not Installed
        Jinja2: 3.1.2
       libgit2: Not Installed
  looseversion: 1.0.2
      M2Crypto: Not Installed
          Mako: Not Installed
       msgpack: 1.0.2
  msgpack-pure: Not Installed
  mysql-python: Not Installed
     packaging: 22.0
     pycparser: 2.21
      pycrypto: Not Installed
  pycryptodome: 3.9.8
        pygit2: Not Installed
  python-gnupg: 0.4.8
        PyYAML: 6.0.1
         PyZMQ: 23.2.0
        relenv: 0.14.2
         smmap: 5.0.1
       timelib: 0.2.4
       Tornado: 4.5.3
           ZMQ: 4.3.4

System Versions:
          dist: debian 12 bookworm
        locale: utf-8
       machine: x86_64
       release: 6.1.0-17-amd64
        system: Linux
       version: Debian GNU/Linux 12 bookworm

```

**Comment by randombash** [comment]
On Debian 11, Intalling  gitdb and gitpython using salt-pip (on top of what I had installed already using apt) helped me with those same errors:

> Mar 05 16:09:34 saltmaster salt-master[2305036]: [ERROR   ] gitfs is configured but could not be loaded, is GitPython installed?
Mar 05 16:09:34 saltmaster salt-master[2305036]: [CRITICAL] No suitable gitfs provider module is installed.

```
root@saltmaster:/usr/bin# salt --versions-report
Salt Version:
          Salt: 3007.1
 
Python Version:
        Python: 3.10.14 (main, Apr  3 2024, 21:30:09) [GCC 11.2.0]
 
Dependency Versions:
          cffi: 1.16.0
      cherrypy: unknown
      dateutil: 2.8.2
     docker-py: Not Installed
         **gitdb: Not Installed
     gitpython: Not Installed**
        Jinja2: 3.1.4
       libgit2: Not Installed
  looseversion: 1.3.0
      M2Crypto: Not Installed
          Mako: Not Installed
       msgpack: 1.0.7
  msgpack-pure: Not Installed
  mysql-python: Not Installed
     packaging: 23.1
     pycparser: 2.21
      pycrypto: Not Installed
  pycryptodome: 3.19.1
        pygit2: Not Installed
  python-gnupg: 0.5.2
        PyYAML: 6.0.1
         PyZMQ: 25.1.2
        relenv: 0.16.0
         smmap: Not Installed
       timelib: 0.3.0
       Tornado: 6.3.3
           ZMQ: 4.3.4
 
Salt Package Information:
  Package Type: onedir
 
System Versions:
          dist: debian 11.11 bullseye
        locale: utf-8
       machine: x86_64
       release: 5.10.0-32-amd64
        system: Linux
       version: Debian GNU/Linux 11.11 bullseye
 
root@saltmaster:/usr/bin#
root@saltmaster:/srv/formulas# **apt list python3-git**
Listing... Done
python3-git/oldstable-security,now 3.1.14-1+deb11u1 all [installed]
N: There is 1 additional version. Please use the '-a' switch to see it
root@saltmaster:/srv/formulas# 
root@saltmaster:/usr/bin# **apt search gitdb**
Sorting... Done
Full Text Search... Done
python3-gitdb/oldstable,now 4.0.5-1 all [installed,automatic]
  pure-Python git object database (Python 3)

root@saltmaster:/usr/bin#
root@saltmaster:/usr/bin# **salt-pip install gitpython**
Collecting gitpython
  Downloading GitPython-3.1.44-py3-none-any.whl.metadata (13 kB)
Collecting gitdb<5,>=4.0.1 (from gitpython)
  Using cached gitdb-4.0.12-py3-none-any.whl.metadata (1.2 kB)
Collecting smmap<6,>=3.0.1 (from gitdb<5,>=4.0.1->gitpython)
  Using cached smmap-5.0.2-py3-none-any.whl.metadata (4.3 kB)
Downloading GitPython-3.1.44-py3-none-any.whl (207 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 207.6/207.6 kB 14.0 MB/s eta 0:00:00
Using cached gitdb-4.0.12-py3-none-any.whl (62 kB)
Using cached smmap-5.0.2-py3-none-any.whl (24 kB)
Installing collected packages: smmap, gitdb, gitpython
Successfully installed gitdb-4.0.12 gitpython-3.1.44 smmap-5.0.2
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv

[notice] A new release of pip is available: 23.3.2 -> 25.0.1
[notice] To update, run: /opt/saltstack/salt/bin/python3.10 -m pip install --upgrade pip
root@saltmaster:/usr/bin# salt-pip install gitdb
Requirement already satisfied: gitdb in /opt/saltstack/salt/extras-3.10 (4.0.12)
Requirement already satisfied: smmap<6,>=3.0.1 in /opt/saltstack/salt/extras-3.10 (from gitdb) (5.0.2)
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv

[notice] A new release of pip is available: 23.3.2 -> 25.0.1
[notice] To update, run: /opt/saltstack/salt/bin/python3.10 -m pip install --upgrade pip
root@saltmaster:/usr/bin# 
```

So now salt --versions-report says
``` 
         gitdb: 4.0.12
     gitpython: 3.1.44
```

And remote formulas are available.
